### PR TITLE
rose app-run: poll delays: ISO8601 duration

### LINKED ISSUE #1552: rose app-run: poll delays should support ISO-8601 durations
Running `rose app-run` with the following:

```cfg
[poll]
delays=0,12*PT15M
# ...
```

currently results in a mysterious error like this:

```
[FAIL] 'M'
```

It turns out that we are not yet supporting ISO-8601 duration for poll delays.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Support ISO8601 duration in poll delays syntax. Improve problem
reporting on problem with legacy syntax.

Fix #1552.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
@benfitzpatrick please review.

**Comment by matthewrmshin** [comment]
(N.B. We can support ISO8601 time recurrence as well, in theory.)

File: lib/python/rose/app_run.py [hunk 1]
```diff
@@ -70,12 +74,11 @@ class PollTimeoutError(Exception):
     """An exception raised when time is out for polling."""
 
     def __str__(self):
-        sec, delay, items = self.args
+        time_point, delay, items = self.args
```
Comment (benfitzpatrick) [inline lib/python/rose/app_run.py]: `duration` is more appropriate.

Comment (benfitzpatrick) [inline lib/python/rose/app_run.py]: I take it back

**Pull request merged**